### PR TITLE
Allow psalm 4

### PR DESCRIPTION
**Pull request by tugmaks**

### CODE REVIEW

**Comment by simPod** [comment]
@tugmaks Hi, are you planning to finish the PR?

**Comment by tugmaks** [comment]
@simPod I tried to fix tests, now they are green, but with some skips

**Comment by enumag** [comment]
ping @weirdan

File: tests/acceptance/QueryBuilderDbal.feature
Comment (simPod) [inline tests/acceptance/QueryBuilderDbal.feature]: redundant line

File: tests/acceptance/QueryBuilderDbal.feature
Comment (simPod) [inline tests/acceptance/QueryBuilderDbal.feature]: redundant line

**Review by simPod — APPROVED** [review]

File: tests/acceptance/QueryBuilderDbal.feature
Comment (tugmaks) [reply tests/acceptance/QueryBuilderDbal.feature]: fixed

File: tests/acceptance/QueryBuilderDbal.feature
Comment (tugmaks) [reply tests/acceptance/QueryBuilderDbal.feature]: fixed

**Comment by enumag** [comment]
I wanted to update so I took this PR + rebased #32 and pushed it all into https://github.com/enumag/doctrine-psalm-plugin (dev-master).

I'm experiencing some new false positives now but not sure how to fix them... I think it will be better to report those when everything is merged here.

**Comment by dkarlovi** [comment]
@weirdan @simPod this passes now, what are the objections to get it merged?

**Comment by simPod** [comment]
@dkarlovi I'm not a maintainer but you have my approval already!

**Comment by dkarlovi** [comment]
@simPod sorry, my bad! :blush: Thanks!

**Comment by desmax** [comment]
@weirdan how can we help to merge this?

**Comment by VincentLanglet** [comment]
> @weirdan how can we help to merge this?

Fixing tests maybe ?

**Comment by weirdan** [comment]
Tests already pass on `allow-psalm-4` branch (this repo, not @tugmaks's, but my branch includes his commits) when using `psalm@dev-master` (https://github.com/weirdan/doctrine-psalm-plugin/runs/1507779961, high deps). What I need now is a new release of Psalm to pin to - because of vimeo/psalm#4782 (and I want vimeo/psalm#4789 as well).

**Comment by enumag** [comment]
Both are merged already so it really is just about new psalm release.

@weirdan Are you planning to merge https://github.com/weirdan/doctrine-psalm-plugin/pull/32 as well?

**Comment by VincentLanglet** [comment]
4.3.2 is released @weirdan :)

**Comment by weirdan** [comment]
Addressed in 1.0.0 release

**Comment by enumag** [comment]
@weirdan I understand that #32 is not addressed in the release then? Since it was not merged... Do you have some plans regarding that?

**Comment by iluuu1994** [comment]
Thank you @weirdan!

**Comment by weirdan** [comment]
> I understand that #32 is not addressed in the release then

Not yet, the build is green there, but it needs additional fixes still.

**Pull request closed without merging**
